src/core/types.ts

```diff
@@ -7,9 +7,12 @@ export type Setter = <Value, Update>(
   update: Update
 ) => void
 
+export type Scope = symbol | string | number
```

Comment:
We could technically support `object`, but for now let's limit it to primitives as I don't see use cases of `object`.

Reply:
@dai-shi I like the symbol solution more than other solutions, I don't think an object is necessary, What's your idea? and for strings or numbers, as you know, they are not unique, maybe libraries use them, and jotai would find the same scope for two or more libraries, and mess them up.

Reply:
I agree that most use cases we have in our mind is just `symbol`s. `string`s and `number`s might be useful for certain cases if they know what they are doing. Like two separate libs share atoms? Don't know. I just think we don't need to limit too much.

Reply:
That's ok, but maybe two libs choose a scope like this:

Reply:
That's ok, but maybe two libs choose a scope like this:
```jsx
// in lib 1
countAtom.scope = 1
// in lib 2 
anotherAtom.scope = 1
```
And yes, I agree with you that we shouldn't limit too much.

Reply:
Yes, I know. That use case might be valid. Thanks for your notice anyway. It's nice to discuss this new feature in deep.

Reply:
Your welcome, I think we can tell the folks in the docs that it's better to use symbols, it's another good idea.

Reply:
Oh, I forgot to add docs, at least something minimal.